### ISSUE TITLE
PropertyAccessor::writeProperty() throws misleading NoSuchPropertyException

### ISSUE THREAD

**Issue by dohpaz42** [symfony/symfony#24742]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | master

I had (incorrectly) architected myself into a situation where I have an entity without a setter for a property, and when trying to hydrate the entity from a FormType, I would get the following exception:

> Could not determine access type for property \"_property_\" in class \"_AppBundle\Entity\MyEntity_\".

I'm not always a smart man, so it wasn't immediately obvious what that message meant. It took me dumping the contents of the `$access` array before it clicked:

> /var/www/vendor/symfony/symfony/src/Symfony/Component/PropertyAccess/PropertyAccessor.php:653:
> array(3) {
>   [0] =>
>   bool(true)
>   [1] =>
>   int(4)
>   [2] =>
>   string(190) "Neither the property "_property_" nor one of the methods "addProperty()"/"removeProperty()", "setProperty()", "property()", "__set()" or "__call()" exist and have public access in class "_AppBundle\Entity\MyEntity_"."
> }

Long story short: I believe the [NoSuchPropertyException that is thrown](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L547) should include the `$access[self::ACCESS_NAME]` if it exists.

Thank you!

**Comment by xabbuh** [comment]
Can you be a bit more specific of what you would have liked to have instead? I am not sure if I completely got that.

**Comment by dohpaz42** [comment]
@xabbuh, essentially, I envision the exception appending the value of `$access[self::ACCESS_NAME]` to the existing message. e.g.,

> Could not determine access type for property "property" in class "AppBundle\Entity\MyEntity": Neither the property "property" nor one of the methods "addProperty()"/"removeProperty()", "setProperty()", "property()", "__set()" or "__call()" exist and have public access in class "AppBundle\Entity\MyEntity"

So, the line of code might look like:

(formatted for readability)

    throw new NoSuchPropertyException(sprintf(
        'Could not determine access type for property "%s" in class "%s"%s.',
        $property,
        get_class($object),
        isset($access[self::ACCESS_NAME]) ? ': ' . $access[self::ACCESS_NAME] : ''
    ));

**Comment by xabbuh** [comment]
Would you like to open a pull request with your suggestion?
